src/main/java/net/mcreator/element/types/interfaces/IGUI.java

```diff
@@ -17,6 +17,19 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-package net.mcreator.element.parts.gui;
+package net.mcreator.element.types.interfaces;
 
-public interface IMachineNamedComponent {}
+import net.mcreator.element.parts.gui.GUIComponent;
+
+import java.util.Collection;
+import java.util.List;
+
+@SuppressWarnings("unused") public interface IGUI {
+
+	List<GUIComponent> getComponents();
+
+	default Collection<GUIComponent> getComponentsOfType(String type) {
+		return getComponents().stream().filter(c -> c.getClass().getName().equals(type)).toList();

```

Comment:
This should actually match simple name of a class because `getClass().getName()` invoked on any `GUIComponent` subclass will return its FQDN. For instance, Button class will give you `net.mcreator.element.parts.gui.Button` whereas you want it to return just `Button`.
```suggestion
		return getComponents().stream().filter(c -> c.getClass().getSimpleName().equals(type)).toList();
```